### ISSUE TITLE
Build failing on gcc 10 due to -Werror=all option

### ISSUE THREAD

**Issue by SoftwareApe** [google/orbit#2135]
With gcc-10 I get the following error:

```log
[ 26%] Building CXX object src/SyntaxHighlighter/CMakeFiles/SyntaxHighlighter.dir/X86Assembly.cpp.o
In file included from /nix/store/pclcc5kl1bkkns8410f6l0dhrnmxll7x-glibc-2.32-37-dev/include/string.h:519,
                 from /home/user/.conan/data/gtest/1.10.0/_/_/package/05a551f43987ffd25a33f0d5949b39b3e26af262/include/gtest/internal/gtest-port.h:250,
                 from /home/user/.conan/data/gtest/1.10.0/_/_/package/05a551f43987ffd25a33f0d5949b39b3e26af262/include/gtest/internal/gtest-internal.h:40,
                 from /home/user/.conan/data/gtest/1.10.0/_/_/package/05a551f43987ffd25a33f0d5949b39b3e26af262/include/gtest/gtest.h:62,
                 from /work/orbit-1.61/src/OrbitBase/OrbitApiTest.cpp:5:
In function ‘char* strncpy(char*, const char*, size_t)’,
    inlined from ‘orbit_api::EncodedEvent::EncodedEvent(orbit_api::EventType, const char*, uint64_t, orbit_api_color)’ at /work/orbit-1.61/src/Api/include/Api/EncodedEvent.h:54:19,
    inlined from ‘virtual void OrbitApi_Encoding_Test::TestBody()’ at /work/orbit-1.61/src/OrbitBase/OrbitApiTest.cpp:25:85:
/nix/store/pclcc5kl1bkkns8410f6l0dhrnmxll7x-glibc-2.32-37-dev/include/bits/string_fortified.h:91:34: error: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ output truncated copying 33 bytes from a string of length 43 [-Werror=stringop-truncation]
   91 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |  ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
make[2]: *** [src/OrbitBase/CMakeFiles/OrbitBaseTests.dir/build.make:180: src/OrbitBase/CMakeFiles/OrbitBaseTests.dir/OrbitApiTest.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

I removed the -Werror=all option and now it builds.

**Comment by SoftwareApe** [comment]
With the flag removed the build seems to still be successful though

```log
100% tests passed, 0 tests failed out of 728

Total Test time (real) =   5.04 sec

The following tests did not run:
         76 - LinuxTracingIntegrationTest.SchedulingSlices (Skipped)
         77 - LinuxTracingIntegrationTest.FunctionCalls (Skipped)
         78 - LinuxTracingIntegrationTest.CallstackSamplesAndAddressInfos (Skipped)
         79 - LinuxTracingIntegrationTest.CallstackSamplesTogetherWithFunctionCalls (Skipped)
         80 - LinuxTracingIntegrationTest.CallstackSamplesWithFramePointers (Skipped)
         81 - LinuxTracingIntegrationTest.CallstackSamplesWithFramePointersTogetherWithFunctionCalls (Skipped)
         82 - LinuxTracingIntegrationTest.ThreadStateSlices (Skipped)
         83 - LinuxTracingIntegrationTest.ThreadNames (Skipped)
         84 - LinuxTracingIntegrationTest.ModuleUpdateOnDlopen (Skipped)
        311 - ServiceUtils.CategoriesTracepoints (Skipped)
        312 - ServiceUtils.NamesTracepoints (Skipped)
```

**Comment by dimitry-** [comment]
look like a duplicate of #1624

**Comment by SoftwareApe** [comment]
@dimitry- true, i didn't find that issue. I used the profile gcc10_relwithdebinfo btw. Not release as mentioned there.

**Comment by dimitry-** [comment]
I think I will just disable stringopt-truncation warning for gcc then.

**Comment by dimitry-** [comment]
should be fixed by #2137
